### PR TITLE
OSDOCS-8429: DNS Improvements

### PR DESCRIPTION
**Pull request by skrthomas**
For peer review/merge review:
Merge to only the no-1.5 branch - no cherrypicks are required.
This PR is part of an experiment for simplifying merges for asynchronous content, and I will open one PR against main to incorporate all of the Network Observability 1.5 content just before its GA.

Version(s):

4.11+
Issue:

https://issues.redhat.com/browse/OSDOCS-8429
Link to docs preview:

DNS Overview (added TCP): https://68507--docspreview.netlify.app/openshift-enterprise/latest/network_observability/observing-network-traffic#network-observability-dns-overview_nw-observe-network-traffic

Working with DNS tracking (removed privileged=true): https://68507--docspreview.netlify.app/openshift-enterprise/latest/network_observability/observing-network-traffic#network-observability-dns-tracking_nw-observe-network-traffic
QE review:
- [X] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/network-observability-dns-tracking.adoc [hunk 1]
```diff
@@ -33,12 +33,15 @@ spec:
...
 . When you refresh the *Network Traffic* page, there are new DNS representations you can choose to view in the *Overview* and *Traffic Flow* views and new filters you can apply.
 .. Select new DNS choices in *Manage panels* to display graphical visualizations and DNS metrics in the *Overview*.
 .. Select new choices in *Manage columns* to add DNS columns to the *Traffic Flows* view.
-.. Filter on specific DNS metrics, such as *DNS Id*, *DNS Latency* and *DNS Response Code*, and see more information from the side panel.
\ No newline at end of file
+.. Filter on specific DNS metrics, such as *DNS Id*, *DNS Latency* and *DNS Response Code*, and see more information from the side panel.
+
+[NOTE]
+===
+TCP handshake packets do not have DNS headers. You can filter for flows that have DNS errors using the "DNSError" filter in the *Common* filters and specifying "0" errors.
```
Comment (Amoghrd) [inline modules/network-observability-dns-tracking.adoc:46]: This statement needs to be updated based on what I have told in the demo. Let me know when once you had a chance to look at the video

File: modules/network-observability-dns-tracking.adoc [hunk 2]
```diff
@@ -33,12 +33,15 @@ spec:
...
 ----
 <1> You can set the `spec.agent.ebpf.features` parameter list to enable DNS tracking of each network flow in the web console.
-<2> Note that the `spec.agent.ebpf.privileged` specification value must be `true` for DNS tracking to be enabled.
 
 . When you refresh the *Network Traffic* page, there are new DNS representations you can choose to view in the *Overview* and *Traffic Flow* views and new filters you can apply.
 .. Select new DNS choices in *Manage panels* to display graphical visualizations and DNS metrics in the *Overview*.
 .. Select new choices in *Manage columns* to add DNS columns to the *Traffic Flows* view.
-.. Filter on specific DNS metrics, such as *DNS Id*, *DNS Latency* and *DNS Response Code*, and see more information from the side panel.
\ No newline at end of file
+.. Filter on specific DNS metrics, such as *DNS Id*, *DNS Latency* and *DNS Response Code*, and see more information from the side panel.
```
[the remark this answers is not in the document]
Comment (Amoghrd) [reply modules/network-observability-dns-tracking.adoc:42]: And one other thing is DNSLatency and ResponseCode columns are enabled by default when DNSTracking feature is enabled. Users need to go to Manage Columns and add the other 2 columns if they wish to do so.
Dont know how to better tell this but😅

File: modules/network-observability-dns-overview.adoc [hunk 3]
```diff
@@ -19,4 +19,4 @@ When DNS tracking is enabled, you can see the following metrics represented in a
 * Top 5 DNS response code
```
Comment (Amoghrd) [inline modules/network-observability-dns-overview.adoc:19]: think this was missed previosuly but this needs to be updated to `Top X` for all the panel names

File: modules/network-observability-dns-tracking.adoc [hunk 4]
```diff
@@ -33,12 +33,15 @@ spec:
     ebpf:
       features:
        - DNSTracking           <1>
-      privileged: true         <2>
```
Comment (Amoghrd) [inline modules/network-observability-dns-tracking.adoc:36]: flowcollector API has to be updated as suggested in RTT doc. Also PascalCase changes needs to be incorporated too

Comment (skrthomas) [reply modules/network-observability-dns-tracking.adoc:36]: @Amoghrd Ok I removed the privileged: true and added the recommendation for set sampling to 1

**Review by mletalie — COMMENTED** [review]
/label peer-review-in-progress

**Comment by mletalie** [comment]
https://68507--docspreview.netlify.app/openshift-enterprise/latest/network_observability/observing-network-traffic

**Comment by skrthomas** [comment]
The previews for this seem not to be working. I'm having preview issues on another PR too...not sure if it has to do with the `no-.5` branch or what.

File: modules/network-observability-dns-tracking.adoc [hunk 5]
```diff
@@ -13,32 +13,37 @@ CPU and memory usage increases are observed in the eBPF agent when this feature
...
+    type: eBPF
     ebpf:
       features:
        - DNSTracking           <1>
-      privileged: true         <2>
+      sampling: 1              <2>
 ----
 <1> You can set the `spec.agent.ebpf.features` parameter list to enable DNS tracking of each network flow in the web console.
-<2> Note that the `spec.agent.ebpf.privileged` specification value must be `true` for DNS tracking to be enabled.
+<2> Optionally, consider setting `sampling` to a value of `1` for more accurate metrics.
```
Comment (mburke5678) [inline modules/network-observability-dns-tracking.adoc:39]:
```suggestion
<2> Optionally, set `sampling` to a value of `1` for more accurate metrics.
```

File: modules/network-observability-dns-overview.adoc [hunk 6]
```diff
@@ -13,10 +13,18 @@ You can configure graphical representation of Domain Name System (DNS) tracking
 
 * Troubleshooting: Debug DNS-related issues by tracing DNS resolution steps, tracking latency, and identifying misconfigurations.
 
-When DNS tracking is enabled, you can see the following metrics represented in a chart in the *Overview*. See the _Additional Resources_ in this section for more information about enabling and working with this view.
+When DNS tracking is enabled, you can see the following non-empty metrics represented in a donut or line chart in the *Overview* by default.
```
Comment (mburke5678) [inline modules/network-observability-dns-overview.adoc:16]: Is "donut" the official name? I don't see it elsewhere in the docs.

Comment (mburke5678) [reply modules/network-observability-dns-overview.adoc:16]: Stating _in the *Overview* by default_ almost sounds like you can configure things to show the metrics elsewhere.
```suggestion
By default, when DNS tracking is enabled, you can see the following non-empty metrics represented in a donut or line chart in the *Overview* page.
```

**Comment by mburke5678** [comment]
@skrthomas   A few nits. Otherwise LGTM

File: modules/network-observability-dns-overview.adoc (hunk 6 above)
Comment (skrthomas) [reply modules/network-observability-dns-overview.adoc:16]: Hmm I can see what you mean. I was trying to convey that the mentioned metrics are visible by default. What I'm not saying is that there are others you can configure to view that aren't shown by default...I will adjust the wording here to clarify that.

File: modules/network-observability-dns-tracking.adoc [hunk 7]
```diff
@@ -13,32 +13,37 @@ CPU and memory usage increases are observed in the eBPF agent when this feature
...
 .. Select new DNS choices in *Manage panels* to display graphical visualizations and DNS metrics in the *Overview*.
 .. Select new choices in *Manage columns* to add DNS columns to the *Traffic Flows* view.
-.. Filter on specific DNS metrics, such as *DNS Id*, *DNS Latency* and *DNS Response Code*, and see more information from the side panel.
\ No newline at end of file
+.. Filter on specific DNS metrics, such as *DNS Id*, *DNS Error* *DNS Latency* and *DNS Response Code*, and see more information from the side panel. The *DNS Latency* and *DNS Response Code* columns are shown by default.
+
+[NOTE]
+===
+TCP handshake packets do not have DNS headers. TCP protocol flows without DNS headers are shown in the traffic flow data with *DNS Latency*, *ID*, and *Response code* values of "n/a". You can filter out flow data to view only flows that have DNS headers using the *Common* filter "DNSError" equal to "0". 
+===
```
Comment (bscott-rh) [inline modules/network-observability-dns-tracking.adoc:49]:
```suggestion
[NOTE]
====
TCP handshake packets do not have DNS headers. TCP protocol flows without DNS headers are shown in the traffic flow data with *DNS Latency*, *ID*, and *Response code* values of "n/a". You can filter out flow data to view only flows that have DNS headers using the *Common* filter "DNSError" equal to "0". 
====
```
Need an extra `=` to fix the formatting here
Screenshot 2024-01-12 at 12 46 17 PM

**Comment by bscott-rh** [comment]
Merge review LGTM.

**Pull request merged**